src/main/java/uk/ac/ebi/biosamples/jsonschemastore/controller/SchemaExporterController.java

```diff
@@ -0,0 +1,23 @@
+package uk.ac.ebi.biosamples.jsonschemastore.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.ac.ebi.biosamples.jsonschemastore.service.JsonSchemaExporter;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "/exporter/schemas", produces = {"application/json"})
```

Comment:
This is not very RESTful.
In REST everything should revolve around the resource, in our case the checklist. We have:

- /schemas - the collection resource of all checklists
- /schemas/<id> - the item resource of a single checklist
- /schemas/<id>/json-schema - the sub-resource of a single checklist's representation as json-schema

This way the links section in the HAL response would include simple links.

I wouldn't change it now, but I would prefer the API to eventually be resource centric.

Reply:
I agree. But as I understand not all operations are not possible to nicely expose in restful way. Happy to change it to restful way and also to see how we can do that.